tiledb/sm/c_api/tiledb.cc

```diff
@@ -2955,8 +2955,68 @@ int32_t tiledb_array_delete_fragments(
   if (sanity_check(ctx) == TILEDB_ERR || sanity_check(ctx, array) == TILEDB_ERR)
     return TILEDB_ERR;
 
-  throw_if_not_ok(array->array_->delete_fragments(
-      tiledb::sm::URI(uri), timestamp_start, timestamp_end));
+  try {
+    array->array_->delete_fragments(
+        tiledb::sm::URI(uri), timestamp_start, timestamp_end);
+  } catch (std::exception& e) {
+    auto st = Status_ArrayError(e.what());
+    LOG_STATUS_NO_RETURN_VALUE(st);
+    save_error(ctx, st);
+    return TILEDB_ERR;
+  }
+
+  return TILEDB_OK;
+}
+
+int32_t tiledb_array_delete_fragments_list(
```

Comment:
Why do we have the API take a URI and not an opened array?

Reply:
We only need one or the other, and the URI is needed to validate `Consistency`. This was discussed some time back, and `array_delete` was reworked to take in _only_ a URI, delegating the responsibility of opening an array, in `MODIFY_EXCLUSIVE` mode, to the C API wrapper (rather than to the user). This new API maintains that same functionality.